### ISSUE TITLE
Array<Option<Gd<_>>> Crashes on Null Pointer in Godot's Editor

### ISSUE THREAD

**Issue by Aeropolian** [godot-rust/gdext#696]
# Description
It is no longer possible for the editor to store a null pointer in a Option<Gd<_>> in an Array exported from Rust. Instead when adding an element to an array or opening an array containing a null element in the editor the editor simply crashes. When opened in the console to see what happens, the following error/crash message is shown:
```
thread '<unnamed>' panicked at C:\Users\...\.cargo\git\checkouts\gdext-76630c89719e160c\cb41673\godot-core\src\builtin\meta\signature.rs:554:5:
in function `EventInspector::parse_property` at parameter [0] of type godot_core::obj::gd::Gd<godot_core::gen::classes::object::re_export::Object>: `Gd` cannot be null: null
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
## Information
- Windows 10
- Godot 4.2.2
- GDExt, latest commit as of May 3rd, 2024
# Example
Minimum working demonstration of the issue.
```rust
#[derive(GodotClass)]
#[class(base=Node)]
pub struct ErrorExample {
    #[export]
    array: Array<Option<Gd<Node>>>,

    base: Base<Node>,
}

#[godot_api]
impl INode for ErrorExample {
    fn init(base: Base<Node>) -> Self {
        Self {
            array: Array::new(),
            base,
        }
    }
}
```
1. Create a scene with the node.
2. Open the array in the inspector, and try to add an element.
3. Editor crashes.

It will crash due to null pointer even though it should become the None variant as it did previously.

**Comment by Bromeon** [comment]
Duplicate of https://github.com/godot-rust/gdext/issues/283?

**Comment by Aeropolian** [comment]
Well I decided to make a completely fresh project and the issue doesn't happen while the identical code breaks in my existing project. With identical configuration in `Cargo.toml`. I did `cargo clean` and it still didn't work in the existing project.

I did some testing by blocking out parts of my existing project. Something to do with an EditorInspectorPlugin I made a while back even though it was for an unrelated type and somehow causing a crash with *all* Arrays but not direct references. I'm baffled as to how that would cause this issue as neither has/uses Arrays nor any of the other types that fail to work in my earlier example. Additionally, that type it was tested against is not used anywhere. It also says it can't handle the types that end up with issues and doesn't execute anything else for them. But Godot crashes because of Arrays as not supporting nulls when that Inspector exists.
